app/javascript/app/components/table.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+function THeader(props) {
+  return (
+    <thead>
+      <tr>
+        {props.config.map((field, i) =>
+          <th key={'header-' + field.header}>
+            {field.header}
+          </th>
+        )}
+      </tr>
+    </thead>
+  );
+}
+
+function Row(props) {
+  return (
+    <tr key={'offer-' + props.offer_id + '-row'}>
+      {props.config.map((field, i) =>
+        <td key={'offer-' + props.offer_id + '-row-' + i}>
+          {field.data(props.offer)}
+        </td>
+      )}
+    </tr>
+  );
+}
+
+class AdminTable extends React.Component {
+  render() {
+    return (
+      <Table striped bordered condensed hover>
+        <THeader config={this.props.config} />
+        <tbody>
+          this.props.offers().map((offer) =>
```

Comment:
So, regardless of little syntax issues, I've found the reason that this doesn't appear to re-render:
If you intend to do the spread/deconstruction/whatever here using `map`, you need to be certain of whether this is a JS object or an Immutable object. `map` doesn't actually work on a JS object.

I changed line 37-38 to
```
{this.props.offers.map((value, key) =>
                                   <Row offer_id={key} offer={value} {...this.props} />)}
```
and added
```
console.log('offers', this.props.offers);
```
at the top of `render`.
It produced the following output
![image](https://user-images.githubusercontent.com/5559014/29373123-371854fc-827b-11e7-92df-62e6814be573.png)
in which you'll notice that render *is* being called twice, as you would expect - once before the offers are added and once after. However, the empty offers list is an Immutable object, and once it's been populated, it's a JS object (which can't use `map`).
[This is all consistent with what you say in (3)]